src/Nethermind/Nethermind.GitBook/RpcAndCliDataProvider.cs

```diff
@@ -0,0 +1,214 @@
+                .Where(t => t.IsInterface && t != typeof(IModule)))
+            {
+                jsonRpcModules.Add(type);
+            }
+            
+            jsonRpcModules.Add( Assembly.Load("Nethermind.Consensus.Clique").GetTypes()
+                .Where(t => typeof(IModule).IsAssignableFrom(t))
+                .First(t => t.IsInterface && t != typeof(IModule)));
+            
+            return jsonRpcModules;
+        }
+        
+        private List<Type> GetCliModules()
+        {
+            Assembly assembly = Assembly.Load("Nethermind.Cli");
+            List<Type> cliModules = new List<Type>();
+
+            foreach (Type type in assembly.GetTypes()
+                .Where(t => typeof(CliModuleBase).IsAssignableFrom(t))
+                .Where(t => t != typeof(CliModuleBase)))
+            {
+                cliModules.Add(type);
+            }
+            return cliModules;
+        }
+
+        private void AddRpcModulesData(List<Type> rpcTypes)
+        {
+            foreach (Type rpcType in rpcTypes)
+            {
+                string rpcModuleName = rpcType.Name.Substring(1).Replace("Module", "").ToLower();
+                
+                MethodInfo[] moduleMethods = rpcType.GetMethods();
+                Dictionary<string, MethodData> methods =
+                    new Dictionary<string, MethodData>();
+                
+                foreach (MethodInfo moduleMethod in moduleMethods)
+                {
+                    string methodName = moduleMethod.Name.Substring(moduleMethod.Name.IndexOf('_') + 1);
+                    JsonRpcMethodAttribute methodAttribute = moduleMethod.GetCustomAttribute<JsonRpcMethodAttribute>();
+                    
+                    MethodData methodData = new MethodData()
+                    {
+                        IsImplemented = methodAttribute?.IsImplemented,
+                        ReturnType = moduleMethod.ReturnType,
+                        Parameters = moduleMethod.GetParameters(),
+                        Description = methodAttribute?.Description,
+                        EdgeCaseHint = methodAttribute?.EdgeCaseHint,
+                        ResponseDescription = methodAttribute?.ResponseDescription,
+                        ExampleResponse = methodAttribute?.ExampleResponse,
+                        InvocationType = InvocationType.JsonRpc
+                    };
+                    
+                    methods.Add(methodName, methodData);
+                }
+                
+                _modulesData.Add(rpcModuleName, methods);
+            }
+        }
+                
+        private void AddCliModulesData(List<Type> cliTypes)
+        {
+            foreach (Type cliType in cliTypes)
+            {
+                MethodInfo[] moduleMethods = cliType.GetMethods().Where(method => method.Name != "GetType" && method.Name != "ToString" && method.Name != "Equals" && method.Name != "GetHashCode").ToArray();
```

Comment:
not critical here but later you retrieve CliFunction / CliProperty attributes so you could use these attributes to filter out any methods that are not part of the module